src/server/json_family.cc

```diff
@@ -1294,7 +1294,12 @@ OpResult<bool> OpSet(const OpArgs& op_args, string_view key, string_view path,
   return operation_result;
 }
 
-io::Result<JsonPathV2, string> ParsePathV2(string_view path) {
+io::Result<JsonPathV2, string> ParsePathV2(string path) {
+  // We expect all valid paths to start with the root selector, otherwise prepend it
+  if (path.rfind("$", 0) == string::npos) {
```

Comment:
1. why rfind? you only want to check that it starts with `$` ?
2. why changing the signature? you can hold a string stack on stack and reassign path to it. this is a corner case.
3. what about `.` ? I think we also support this syntax for some cases. are you sure this change handles this correctly?

Reply:
1. rfind(0) is starts_with in C++ 17 language
2. ok, changed it, it's just more code and the allocation is likely very cheap
3. fixed